gradio/components/file_explorer.py

```diff
@@ -0,0 +1,215 @@
+"""gr.FileExplorer() component"""
+
+from __future__ import annotations
+
+import itertools
+import os
+import re
+from glob import glob as glob_func
+from pathlib import Path
+from typing import Callable, Literal
+
+from gradio_client.documentation import document, set_documentation_group
+from gradio_client.serializing import JSONSerializable
+
+from gradio.components.base import IOComponent, server
+from gradio.events import (
+    Changeable,
+    EventListenerMethod,
+)
+
+set_documentation_group("component")
+
+
+@document()
+class FileExplorer(Changeable, IOComponent, JSONSerializable):
+    """
+    Creates a file component that allows uploading generic file (when used as an input) and or displaying generic files (output).
+    Preprocessing: passes the uploaded file as a {tempfile._TemporaryFileWrapper} or {List[tempfile._TemporaryFileWrapper]} depending on `file_count` (or a {bytes}/{List{bytes}} depending on `type`)
+    Postprocessing: expects function to return a {str} path to a file, or {List[str]} consisting of paths to files.
+    Examples-format: a {str} path to a local file that populates the component.
+    Demos: zip_to_json, zip_files
+    """
+
+    def __init__(
+        self,
```

Comment:
I actually think we should reorder the parameters in this component to put `glob` first, as its the most common one for users to tweak, and we should make it possible for users to do something like `gr.FileExplorer('.')`. (Similar to e.g. the Dropdown component having `choices` as the first parameter)